Symfony/CS/Fixer/Contrib/ClassKeywordRemoveFixer.php

```diff
@@ -78,12 +85,17 @@ private function replaceClassKeyword(Tokens $tokens, $classIndex)
         while ($tokens[--$classBeginIndex]->isGivenKind(array(T_NS_SEPARATOR, T_STRING)));
         ++$classBeginIndex;
         $classString = $tokens->generatePartialCode($classBeginIndex, $classEndIndex);
-        $classImport = current(array_filter($this->imports, function ($import) use ($classString) {
+        $classImport = current(array_filter($this->imports, function ($import, $alias) use ($classString) {
+            // If the ::class is targeting an alias, just return the corresponding import.
+            if ($classString === $alias) {
+                return $import;
+            }
+
             $classStringArray = explode('\\', $classString);
             $namespaceToTest = $classStringArray[0];
 
             return 0 === strcmp($namespaceToTest, substr($import, -strlen($namespaceToTest)));
-        }));
+        }, ARRAY_FILTER_USE_BOTH));
```

Comment:
`ARRAY_FILTER_USE_BOTH`
was added in PHP 5.6, so you cannot use it on the min. supported version of PHP 5.3, sorry